common/frame.go

```diff
@@ -315,27 +315,51 @@ func (f *Frame) defaultTimeout() time.Duration {
 func (f *Frame) document() (*ElementHandle, error) {
 	f.log.Debugf("Frame:document", "fid:%s furl:%q", f.ID(), f.URL())
 
-	rt := k6common.GetRuntime(f.ctx)
-	if f.documentHandle != nil {
-		return f.documentHandle, nil
+	if cdh, ok := f.cachedDocumentHandle(); ok {
+		return cdh, nil
 	}
 
 	f.waitForExecutionContext(mainWorld)
 
+	dh, err := f.newDocumentHandle()
+	if err != nil {
+		return nil, fmt.Errorf("newDocumentHandle: %w", err)
+	}
+	f.executionContextMu.Lock()
+	defer f.executionContextMu.Unlock()
+	f.documentHandle = dh
+
+	return dh, nil
+}
+
+func (f *Frame) cachedDocumentHandle() (*ElementHandle, bool) {
+	f.executionContextMu.RLock()
+	defer f.executionContextMu.RUnlock()
+
+	return f.documentHandle, f.documentHandle != nil
+}
+
+func (f *Frame) newDocumentHandle() (*ElementHandle, error) {
+	rt := k6common.GetRuntime(f.ctx)
+	doc := rt.ToValue("document")
+
 	opts := evalOptions{
 		forceCallable: false,
 		returnByValue: false,
 	}
-	result, err := f.evaluate(f.ctx, mainWorld, opts, rt.ToValue("document"))
+	result, err := f.evaluate(f.ctx, mainWorld, opts, doc)
```

Comment:
What's the reason for declaring `doc`?

Reply:
A style issue, I thought I made it simpler 🤷 Resolved by: https://github.com/grafana/xk6-browser/pull/263/commits/cb452185a6e50247179e7b92dcdf6b3146fb6340